core/resource/handler/v1beta1/resource.go

```diff
@@ -242,6 +243,34 @@ func (rh ResourceHandler) UpdateResource(ctx context.Context, req *pb.UpdateReso
 	return &pb.UpdateResourceResponse{}, nil
 }
 
+func (rh ResourceHandler) ChangeResourceNamespace(ctx context.Context, req *pb.ChangeResourceNamespaceRequest) (*pb.ChangeResourceNamespaceResponse, error) {
+	tnnt, err := tenant.NewTenant(req.GetProjectName(), req.GetNamespaceName())
+	if err != nil {
+		return nil, errors.GRPCErr(err, "failed to adapt to existing tenant details")
+	}
+
+	newTnnt, err := tenant.NewTenant(req.GetProjectName(), req.GetNewNamespaceName())
+	if err != nil {
+		return nil, errors.GRPCErr(err, "failed to adapt to new tenant")
+	}
+
+	store, err := resource.FromStringToStore(req.GetDatastoreName())
+	if err != nil {
+		return nil, errors.GRPCErr(err, "invalid Datastore Name")
+	}
+
+	response, err := rh.service.Get(ctx, tnnt, store, req.GetResourceName())
```

Comment:
Please put this as part of the `service.ChangeNamespace`. Try to minimize business flow as much as possible in this handler.

Reply:
sure